util/testutil/workers/containerd.go

```diff
@@ -185,22 +190,28 @@ disabled_plugins = ["cri"]
 	if err != nil {
 		return nil, nil, err
 	}
-	if err := integration.WaitUnix(address, 10*time.Second, cmd); err != nil {
+	if err := integration.WaitSocket(address, 10*time.Second, cmd); err != nil {
 		ctdStop()
 		return nil, nil, errors.Wrapf(err, "containerd did not start up: %s", integration.FormatLogs(cfg.Logs))
 	}
 	deferF.Append(ctdStop)
 
-	buildkitdArgs := append([]string{"buildkitd",
-		"--oci-worker=false",
+	// handles only windows case, no effect on unix
+	address = normalizeAddress(address)
+	buildkitdArgs := []string{
+		"buildkitd",
 		"--containerd-worker-gc=false",
 		"--containerd-worker=true",
 		"--containerd-worker-addr", address,
 		"--containerd-worker-labels=org.mobyproject.buildkit.worker.sandbox=true", // Include use of --containerd-worker-labels to trigger https://github.com/moby/buildkit/pull/603
-	}, snBuildkitdArgs...)
+	}
+	if ociWorkerFlag != "" {
```

Comment:
Let's try something like:

in `util_unix.go`:

```go
func applyBuildkitdPlatformFlags(args []string) []string {
    return append(args, "--oci-worker=false")
}
```

In `util_windows.go`:

```go
func applyBuildkitdPlatformFlags(args []string) []string {
    return args
```

And here you could remove the if and just:

```go
buildkitdArgs = applyBuildkitdPlatformFlags(buildkitdArgs)
```

The idea is that we may have multiple platform specific flags in the future on both Windows and Linux that we will need to account for. It will be easier to edit a specific platform specific file to add/remove them than to keep adding `if` statements here.

Reply:
Sure, much cleaner. Fixed.